labrad/protocol.py

```diff
@@ -424,19 +518,25 @@ def do_connect():
     @inlineCallbacks
     def start_tls(p, cert_string=None):
         try:
-            resp = yield p.sendRequest(C.MANAGER_ID, [(1L, ('STARTTLS', host))])
+            cert = yield p._sendManagerRequest(1, ('STARTTLS', host))
         except Exception:
             raise Exception(
                 'Failed sending STARTTLS command to server. You should update '
                 'the manager and configure it to support encryption or else '
                 'disable encryption for clients. See '
                 'https://github.com/labrad/pylabrad/blob/master/CONFIG.md')
-        cert = resp[0][1]
         p.transport.startTLS(crypto.tls_options(host, cert_string=cert_string))
         returnValue(cert)
 
+    @inlineCallbacks
     def ping(p):
-        return p.sendRequest(C.MANAGER_ID, [(2L, 'PING')])
+        resp = yield p._sendManagerRequest(2, 'PING')
+        features = resp.partition(':')[-1]
```

Comment:
Previously, the ping response was a single string "PONG". I changed this in the manager to look like "PONG:feature1,feature2,..." to keep the type the same, but in fact we never did anything with this response in the past, so we could certainly change the type. Would just have to check here if resp is a string, then it's an old manager, or if resp is a list, then it's a new manager and interpret resp as list of features. I'm fine either way; do you have a preference?